Using TensorFlow backend.
WARNING:root:Running with configuration:
{'alter_retrain_full_training': 0,
 'attack_succ_threshold': 0.85,
 'benign_poison_ratio': 0.001,
 'classifier': 'mlp',
 'clean_ratio': 0.02,
 'convert_mask_to_binary': 0,
 'dataset': 'apg',
 'debug': True,
 'delta_size': 30,
 'device': '0',
 'lambda_1': 0.001,
 'limited_data': 1.0,
 'mask_expand_type': 0,
 'mask_optim_step': 30,
 'mask_size_upperbound': 0,
 'max_iter': 40,
 'mlp_batch_size': 128,
 'mlp_dropout': 0.2,
 'mlp_epochs': 30,
 'mlp_hidden': '1024',
 'mlp_lr': 0.001,
 'mntd_half_training': 0,
 'num_of_train_batches': 5,
 'num_triggers': 1,
 'param_v': 1.0,
 'poison_mal_benign_rate': 0.0,
 'random_state': 42,
 'realizable_only': 0,
 'remain_benign_rate': 1.0,
 'setting': '',
 'subset_benign_rate': 5.0,
 'subset_family': 'plankton',
 'use_last_weight': 1}
DEBUG: 02-27 18:40:39: models.py:242 * 139527157065536 Loading model from models/apg/mlp/mlp_10000-1024-1_lr0.001_b128_e30_d0.2_r42.p...
INFO: 02-27 18:40:47: subset_backdoor_main.py:450 * 139527157065536 X_train: (100187, 10000), X_test: (49347, 10000)
INFO: 02-27 18:40:47: subset_backdoor_main.py:451 * 139527157065536 y_train: (100187,), y_test: (49347,)
INFO: 02-27 18:40:47: subset_backdoor_main.py:452 * 139527157065536 y_train: Counter({0: 90288, 1: 9899}), y_test: Counter({0: 44471, 1: 4876})
INFO: 02-27 18:40:48: subset_backdoor_main.py:335 * 139527157065536 subset size: 40
DEBUG: 02-27 18:40:48: subset_backdoor_main.py:336 * 139527157065536 subset_idx_array first 20: [ 2913  5237 10052 12422 16635 18761 23742 32890 38273 38972 39265 40651
 40985 43633 57638 59955 62828 75997 76492 78880]
DEBUG: 02-27 18:40:51: subset_backdoor_main.py:356 * 139527157065536 subset_train_idxs first 20 (maybe < 20): [53239, 31964, 81485, 86327, 78195, 90663, 47661, 97975, 6411, 28035, 88440, 66405, 72918, 93321, 88896, 15308, 39986, 30935, 38030, 50176]
DEBUG: 02-27 18:40:51: subset_backdoor_main.py:357 * 139527157065536 subset_test_idxs first 20 (maybe < 20): [4804, 42514, 35129, 41121]
INFO: 02-27 18:40:51: subset_backdoor_main.py:361 * 139527157065536 X_subset: (40, 10000), type: <class 'scipy.sparse.csr.csr_matrix'>
DEBUG: 02-27 18:40:51: subset_backdoor_main.py:364 * 139527157065536 no. of samples of subset in training: 36, from testing: 4
DEBUG: 02-27 18:40:51: attrs.py:77 * 139527157065536 Creating converter from 3 to 5
Traceback (most recent call last):
  File "problem_space_with_side_effect_fea.py", line 251, in <module>
    main()
  File "problem_space_with_side_effect_fea.py", line 83, in main
    random_state, subset_family)
  File "/home/puya/Desktop/jigsawpuzzle/subset_backdoor_main.py", line 367, in separate_subset_malware
    clean_model = load_model(CLEAN_MODEL_PATH)
  File "/home/puya/anaconda3/envs/backdoor/lib/python3.6/site-packages/keras/engine/saving.py", line 458, in load_wrapper
    return load_function(*args, **kwargs)
  File "/home/puya/anaconda3/envs/backdoor/lib/python3.6/site-packages/keras/engine/saving.py", line 550, in load_model
    model = _deserialize_model(h5dict, custom_objects, compile)
  File "/home/puya/anaconda3/envs/backdoor/lib/python3.6/site-packages/keras/engine/saving.py", line 242, in _deserialize_model
    model_config = json.loads(model_config.decode('utf-8'))
AttributeError: 'str' object has no attribute 'decode'
